### PR TITLE
[qa] Add ansible linter to build #91

### PR DESCRIPTION
**Pull request by ppabcd**
Add ansible linter and fix Trailing whitespace.
Fixes #19

### CODE REVIEW

**Comment by hispanico** [comment]
This is the ansible-lint output in gitlab-ci:
```
$ ansible-lint tests/test.yml
ANSIBLE0006 supervisorctl used in place of supervisorctl module
/root/.ansible/roles/role_under_test/handlers/main.yml:9
Task/Handler: reload supervisor
```
The main idea is to run ansible-lint before the build of each distribution and made the lint check with the ansible-galaxy rules:
https://github.com/ansible/galaxy-lint-rules
https://robertdebock.nl/2018/10/20/galaxy-lint.html

**Comment by ppabcd** [comment]
i will try with galaxy-ansible-lint

**Comment by ppabcd** [comment]
i've change any file in tasks, when i see ansible-galaxy score do not change. Finally when runtest i get this errors.
@hispanico  @nemesisdesign 
gambar

File: tasks/apt.yml [hunk 1]
```diff
@@ -15,7 +15,6 @@
       - libffi-dev
       - python-dev
       - redis-server
-    state: latest
```
Comment (nemesisdesign) [inline tasks/apt.yml]: why is this removed? 
Is this considered a bad practice? CC @hispanico

File: tasks/multitenancy_migration.yml [hunk 2]
```diff
@@ -115,11 +114,6 @@
     dest: "{{ openwisp2_path }}/multitenancy_prepare_db.py"
     mode: 0754
 
-- name: prepare DB for multitenancy migration
```
Comment (nemesisdesign) [inline tasks/multitenancy_migration.yml]: why are you removing this?

File: .travis.yml [hunk 3]
```diff
@@ -34,6 +34,19 @@ script:
...
+  # run ansible-lint
+  - pip install --user ansible
+  - ansible-galaxy install -r tests/requirements.yml
+  - pip install --user ansible-lint
+  - ln -s $TRAVIS_BUILD_DIR ~/.ansible/roles/role_under_test
+  - git clone https://github.com/ansible/galaxy-lint-rules.git
+  - mv galaxy-lint-rules ~/galaxy
+  - ansible-lint -r ~/galaxy/rules tests/test.yml
+    --exclude ~/.ansible/roles/Stouts.postfix
+    -x 404 -x 405
```
Comment (nemesisdesign) [inline .travis.yml]: what are `404` and `405`? Is there a way to have a general configuration for ignored files and warnings?

Comment (nemesisdesign) [reply .travis.yml]: yes, please use this: https://github.com/ansible/ansible-lint#configuration-file

File: tasks/apt.yml (hunk 1 above)
Comment (ppabcd) [reply tasks/apt.yml]: i've get linter error for state latest and i do deleted all latest state

Comment (ppabcd) [reply tasks/apt.yml]: 30 E403:  Package installs should not use latest

File: tasks/multitenancy_migration.yml (hunk 2 above)
Comment (ppabcd) [reply tasks/multitenancy_migration.yml]: E301:  Commands should not change things if nothing needs doing

File: .travis.yml (hunk 3 above)
Comment (ppabcd) [reply .travis.yml]: 404 and 405 is errors about connection and i disabled it.

Comment (ppabcd) [reply .travis.yml]: 404 Doesn't need a relative path in role

Comment (ppabcd) [reply .travis.yml]: i'll try using configuration file

**Comment by ppabcd** [comment]
@nemesisdesign Always get error because 
```
-subj "/C={{ openwisp2_ssl_country }}/ST={{ openwisp2_ssl_state }}/L={{ openwisp2_ssl_locality }}/O={{ openwisp2_ssl_organization }}/CN={{ openwisp2_ssl_common_name }}" \
```
is longer than 120 chars

**Comment by nemesisdesign** [comment]
PS: great job BTW! :+1:  :+1:

**Comment by dwang** [comment]
@nemesisdesign @ppabcd 

Should be ready for review now. 
Any other things to fix? I was thinking of adding back `state: latest` to the `apt` and `yum` roles since the lint rule is already being skipped.

**Comment by dwang** [comment]
As of now, two lint rules are being skipped:

- 303 (CommandsInsteadOfModulesRule) - There is no `reload` state in the Ansible supervisorctl module.
- 404 (PackageIsNotLatestRule) - `state=latest` is needed for some packages in `apt`, `yum`, and `pip`.
